### ISSUE TITLE
Feature Request: auto import raw *.fit files from dropbox folder

### ISSUE THREAD

**Issue by funkymaster** [GoldenCheetah/GoldenCheetah#2710]
Hi, it would be nice if we could auto import *raw data files* from a dropbox cloud folder and not only from a local folder. I could mount one dropbox account with my raw *.fit files to my local drive and auto import it then into my athlete profile. As far as I know it is unfortunately not possible to mount a second dropbox account on my pc and therefore not possible to manage a second or more athletes in this way. By the automatic import from a dropbox cloud folder, this topic could be solved.

**Comment by funkymaster** [comment]
Hello, are you planning to work on this feature in the near future? Actually everything that is needed is already there. Because Dropbox synchronisation and auto-import already exist. It would only be necessary to extend the Dropbox synchronisation with the import of fit files or to extend the automatic import function with the import from a Dropbox cloud folder. Interfaces and services should therefore already exist. Could someone please combine these functions and create this new feature? Many thanks.

**Comment by amtriathlon** [comment]
The parking lot milestone indicates nobody is planning to work on this feature for now.
Since you already have analyzed the problem and, apparently concluded it is easy to solve, may be you can implement the solution by yourself.
Once you are are done, please make a pull request so other users can benefit from your work.

**Comment by funkymaster** [comment]
Sorry, I am unable to implement a solution myself but I want to share a workaround for it.

In order to be able to use several Dropbox accounts at the same time, I set up an LXC container on a Debian server with a separate user for each athlete.
Under Linux, it is possible with a headless CLI installation to have several Dropbox accounts running in the background at the same time on one PC. In my case in the LXC container.
The Dropbox user directories are provided via a Samba share then.
GC can now import the raw *.fit files from the mapped network drive.
For users without their own server, it should also be possible to implement this as a desktop solution. Either with a Linux PC or on Windows in a virtual machine with Linux.

**Comment by paulj49457** [comment]
If this is still relevant & useful I'm happy to look at implementing it, but in the Athlete -> Auto Import dialogue which of the following is required/acceptable?

i) An overall checkbox to run the "Check for new Activities" functionality on all cloud services, or

ii) Individual options to add an existing cloud account(s) (Dropbox/Strava/RideWithGPS, etc) instead of a local directory, and for it run the  "Check for new Activities" for that specific cloud account ?

I expect the i) will be simple and quick to implement.

**Comment by funkymaster** [comment]
Yes, this is still relevant.

ii) sounds great and I'd prefer it.
Maybe someone, like in my case, does want to import from Dropbox but not from other services.

**Comment by paulj49457** [comment]
Looks like this functionality already exists:

https://github.com/GoldenCheetah/GoldenCheetah/pull/4357#issuecomment-1527396979

**Comment by funkymaster** [comment]
Yes, I know this function exists, but import only seems to work with *.json files.
Maybe I'm doing something wrong, but I can't sync/import raw Cycling Computer *.fit files there.
This works for me currently only with the auto import function.

**Comment by amtriathlon** [comment]
Dropbox cloud service uses .json files to preserve all data on sync, but cloud services base class supports .fit files as explained in https://github.com/GoldenCheetah/GoldenCheetah/wiki/Using-the-Cloud-Services-API#file-formats-and-compression
To add a file format parameter to file cloud services (Dropbox and Local) would be a better alternative than auto-import, IMHO.

**Comment by amtriathlon** [comment]
I tested this using FIT files uploaded by GC, it works for download only provided files are named according GC convention, for example: 2024_12_29_08_43_58.fit.zip
